### PR TITLE
Be able to add AbilityExtensions in Bot constructor

### PR DESCRIPTION
**Pull request by christianblos**
Currently it's only possible to add ability extensions by defining a method that returns an AbilityExtension. This PR allows to add extensions in the Bot constructor already. This is especially useful when using Spring Dependency Injection.

Together with my other Pull Requests [#810](https://github.com/rubenlagus/TelegramBots/pull/810) and [#812](https://github.com/rubenlagus/TelegramBots/pull/812) you can now create a Bot like this:

```
@Component
public class MyBot extends AbilityBot {

	private final MyBotConfig config;

	public MyBot(MyBotConfig config, List<AbilityExtension> extensions) {
		super(config.getToken(), config.getUsername(), MapDBContext.onlineInstance(config.getDbFile()));
		this.config = config;
		this.extensions.addAll(extensions);
	}

	@Override
	public int creatorId() {
		return config.getCreatorUserId();
	}

}
```

All ability extensions are now registered automatically if they are configured as a Spring Bean.

### CODE REVIEW

File: telegrambots-abilities/src/main/java/org/telegram/abilitybots/api/bot/BaseAbilityBot.java [hunk 1]
```diff
@@ -106,6 +106,7 @@
     private final String botUsername;
 
     // Ability registry
+    protected final List<AbilityExtension> extensions = new ArrayList<>();
```
Comment (addo37) [inline telegrambots-abilities/src/main/java/org/telegram/abilitybots/api/bot/BaseAbilityBot.java:109]: I find that exposing this (and the pattern of usage) is not optimal. We could keep this a private list and expose a "addExtensions" method that will add the extension.

```java
public addExtension(List extensions) ....`

**Review by addo37 — CHANGES_REQUESTED** [review]
Again, thanks for the changes.

File: telegrambots-abilities/src/main/java/org/telegram/abilitybots/api/bot/BaseAbilityBot.java (hunk 1 above)
Comment (christianblos) [reply telegrambots-abilities/src/main/java/org/telegram/abilitybots/api/bot/BaseAbilityBot.java:109]: Good point, I also thought about this. Will change it 👍

**Comment by christianblos** [comment]
I updated the wiki and test as well. Note that this PR requires #812 to be merged first! The ExtensionTest is currently failing because the abilities are still registered in the constructor. As soon as #812 is merged and the abilities are loaded later, the test will succeed.

**Comment by rubenlagus** [comment]
Same here, please move to dev branch

**Comment by christianblos** [comment]
Target is now the dev branch

**Pull request merged**